packages/components/label/src/label.tsx

```diff
@@ -0,0 +1,79 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { invariant } from '@commercetools-uikit/utils';
+import Text from '@commercetools-uikit/text';
+import RequiredIndicator from './required-indicator';
+
+type TBasicTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+type TLabelProps = {
+  /**
+   * Use this to set the relationships between the label and a non-form element.
+   * For example:
+   *   <label id="foo">
+   *   <div aria-labelledby="foo">
+   */
+  id?: string;
+  /**
+   * Use this to set the relationships between the label and a form element.
+   * For example:
+   *   <label for="foo">
+   *   <input id="foo">
+   *
+   *   <label id="foo">
+   *   <input aria-labelledby="foo">
+   */
+  htmlFor?: string;
+  isBold?: boolean;
+  isRequiredIndicatorVisible?: boolean;
+  tone?: 'primary' | 'inverted';
+} & TBasicTextProps;
+
+const Label = (props: TLabelProps) => {
+  invariant(
+    !(Boolean(props.id) && Boolean(props.htmlFor)),
+    `ui-kit/Label: provide only the "id" or the "htmlFor" properties, not both.`
+  );
+
+  const hasContent =
+    props.intlMessage || Boolean(React.Children.count(props.children));
+
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`intlMessage\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`children\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+
+  return (
+    <label id={props.id} htmlFor={props.htmlFor}>
+      <Text.Body tone={props.tone} isBold={props.isBold}>
+        {props.intlMessage ? (
+          <FormattedMessage {...props.intlMessage} />
+        ) : (
+          props.children
+        )}
+        {props.isRequiredIndicatorVisible && <RequiredIndicator />}
+        {props.isRequiredIndicatorVisible && <RequiredIndicator />}
+      </Text.Body>
```

Comment:
Can't we maybe render the text as `span` and render the indicator next to it (instead of inside)?

PS: also, you have duplicated lines here.

Reply:
🤦